### PR TITLE
Include plugin version in manifest of instant builds

### PR DESCRIPTION
**Pull request by thanwah**
Add a <meta-data> element to the <application> element of the Android
manifest indicating which version of the plugin was used.
This is only included for instant app builds.

Example:
<meta-data android:name="play-instant-unity-plugin.version" android:value="1.0"/>

### CODE REVIEW

**Comment by jamcohen** [comment]
LGTM!

**Comment by thanwah** [comment]
Hey @jamcohen, this wasn't quite ready for review before now, since the original code you reviewed only supported Unity 2018+. Can you please re-review with the pre-2018 changes?

File: GooglePlayInstant/Editor/PlayInstantBuilder.cs [hunk 1]
```diff
@@ -212,6 +214,13 @@ public static bool BuildAndSign(BuildPlayerOptions buildPlayerOptions)
         /// <returns>True if the build succeeded, false if it failed or was cancelled.</returns>
         public static bool Build(BuildPlayerOptions buildPlayerOptions)
         {
+            var checkInstantManifestResult = AndroidManifestHelper.GetAndroidManifestUpdater().CheckInstantManifest();
```
Comment (jamcohen) [inline GooglePlayInstant/Editor/PlayInstantBuilder.cs]: If we're already checking the version here, should we instead call SwitchToInstant here? I'm thinking about cases where we update the plugin with some new AndroidManifest requirement, but their manifest doesn't get rebuilt because their project is already in instant mode. It would also make the Legacy codepath more in line with the IPostGenerateGradleAndroidProject method.

Comment (thanwah) [reply GooglePlayInstant/Editor/PlayInstantBuilder.cs]: Good question. I also considered calling SwitchToInstant here.

The really nice thing about IPostGenerateGradleAndroidProject on 2018+ is that we apply our changes to the fully merged AndroidManifest.xml right before the build. As far as I know, it always works.

The pre-2018 approach where we apply changes to one of possibly many manifests pre-merge is more brittle, and I've heard several people from other Google Unity projects say as much. I expect that there may be cases where after the developer saves the manifest in Instant mode, that they have to go in and manually modify the manifest for some reason or another.

My thinking here was to perform the least invasive change possible - just update the plugin version. Also, I only save the file if it needs an update, which avoids file I/O and potentially causing Unity to need to refresh assets.

In short - this approach is a bit more code, but seems less risky to me.

Comment (thanwah) [reply GooglePlayInstant/Editor/PlayInstantBuilder.cs]: Yes, that's fair. We could check the file for correctness and at least prompt a warning if anything seems amiss. I will plan to save that as future work.
